### ISSUE TITLE
"Speack Typed Characters, NVDA+2" can not work while enabled as using Microsoft New Phonetic IME in Microsoft Office Word Tranditional Chinese version

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1382]
**Reported by james on 2011-02-21 07:36**
As testing the new NVDA version 2011.2 beta2 on the Window XP Traditional Chinese version, it was found that the key command "Speack Typed Characters, NVDA+2" cannot work while typing Traditional Chinese characters using Microsoft New Phonetic IME in Windows Office Word. However, if typing English character, it works well.

To reproduce the issue, enable the Speack Typed Characters function by pressing NVDA+2 first. Open the Mircosoft Office Word and then change the input mehtod to Microsoft New Phonetic IME. Finally, type any Chinese character to see if it is spoke out.

This issue was also checked while typing traditional characters in the notepad. There's no problem while using notepad. However, if typing charater in the web page under the browser IE 8.0, the same problem occurred also.

**Comment by bhavyashah** [comment]
@vgjh2005 Could you please test and let us know whether this issue still exists? If not, I suggest closing this ticket as worksforme. @ehollig

**Comment by ehollig** [comment]
Just a friendly reminder for @vgjh2005 to test this issue as requested in https://github.com/nvaccess/nvda/issues/1382#issuecomment-320511173

**Comment by vgjh2005** [comment]
Hi:
It's still a bug.

**Comment by feerrenrut** [comment]
Thanks for the update @vgjh2005. Are you still using windows XP to test this? Support for XP has now been dropped, NVDA 2017.4 will require windows 7 Service Pack 1.

**Comment by vgjh2005** [comment]
Hi:
My system is windows 10 1703 64bit.

2017-10-18 15:19 GMT+08:00, Reef Turner <notifications@github.com>:

**Comment by feerrenrut** [comment]
Not sure if this matches the original steps to reproduce this issue, but here are my findings:

## Software Versions
* Windows 10 version `1703 build 15063.674`
* Microsoft Office 2013
* NVDA `next-14535,3aad54e0`

## To reproduce
1. Open language preferences in the windows 10 settings, if necessary add Chinese (simplified)
1. In the system tray, set the input language to Chinese (simplified)
1. Open MS Word.
1. Ensure that Speak typed characters is turned on using `NVDA+2`
1. Press the `d` key, NVDA speaks `d`, visually a popup shows the character `d` and several Chinese characters with numbers next to them.
1. Press the `1` key, the first Chinese character is added to the document. NVDA speaks `Chinese character`
1. Press the left arrow, NVDA pronounces the Chinese character. 

## Expected
To be able to hear the chinese pronunciation of the characters in the popup, and to hear the character when it is entered into the document.

**Comment by feerrenrut** [comment]
I assume I tested this with espeak, it would be good to confirm that espeak itself can speak these characters outside of word. I would also suggest trying some other synthesizers. If we are sure that the synth can support the characters, then we need to investigate how to get the information out of the popup window.

**Comment by Adriani90** [comment]
@larry801, do you have any further test results / ideas as a chinese speaker? Or do you know of other people who could test this under NVDA 2019.2 RC2? Thanks.

**Comment by Adriani90** [comment]
cc: @cary-rowen could you please test if this issue is still occuring in NVDA last alpha version or NVDA 2023.1?

**Comment by cary-rowen** [comment]
At least in the Simplified Chinese environment, NVDA's support for Chinese input methods cannot meet user needs.
So we specially developed the Chinese input method improvement add-on, currently this add-on is a must-install add-on for NVDA Chinese users, I hope that one day the feature of this add-on will appear in NVDA core, but it may not be easy.
I don't even understand what this Issue is talking about, so I may not be able to provide more information.
Chinese input method improvement add-on:
https://github.com/nvdacn/ime_expressive

**Comment by Adriani90** [comment]
Actually this issue is about NVDA reporting the characters while you are typing. The setting can be turned on in NVDA keyboard settings.

Regarding the addon, if the developers are interested they certainly could raise a pull request and then start a discussion with NV Access on how feasible it is to get it into the core.

**Comment by josephsl** [comment]
Hi,

I assume this is also a problem in Windows 11? Also, could you get IME Expressive add-on into NV Access ad-on store for visibility and discoverability by Chinese IME users across the world?

Thanks.

**Comment by cary-rowen** [comment]
Hi @josephsl @Adriani90 

This add-on was specially developed to meet the Chinese character input needs of Chinese users, and may not meet the coding style of NVDA, at least currently.
What confuses me even more is whether this add-on will negatively impact other CJK users.
Also, I will be listing this add-on in the add-on store in the near future.

Best
Cary

**Comment by cary-rowen** [comment]
I recommend close this:
1. The software version in the original description is too old.
2. I don’t think NVDA+2 should be used to meet Chinese input needs.
Because entering a Chinese character may require multiple steps, this is not a situation that can be covered by a option. See [IME Expressive](https://github.com/nvdacn/ime_expressive)

**Comment by josephsl** [comment]
Hi,

n case IME Expressive would be a useful addition for the Core, let us know so we can go over the add-on code and suggest changes.

Thanks.
